test/twitter/twits_test.exs

```diff
@@ -0,0 +1,99 @@
+defmodule Twitter.TwitsTest do
+  use TwitterWeb.ConnCase
+  # use Twitter.DataCase, async: true
+  # alias Twitter.Accounts.User
+  alias Twitter.Twits
+
+  describe "posts" do
+    alias Twitter.Twits.Post
+
+    @valid_attrs %{body: "some body"}
+    @update_attrs %{body: "some updated body"}
+    @invalid_attrs %{body: nil}
+
+    setup %{conn: conn} do
+      user = user_fixture()
+      conn = init_test_session(conn, current_user: user, user_id: user.id)
+      %{conn: conn, user: user}
+    end
+
+    test "list_posts/1 returns all posts of current user", %{user: user} do
+      post = post_fixture(user)
+      assert post = Twits.list_posts(user)
+    end
+
+    test "list_posts_by_author_id/1 returns all posts of a given author", %{user: user} do
+      # equivalent to list_posts when own id is passed
+      post = post_fixture(user)
+      assert post = Twits.list_posts_by_author_id(user.id)
+    end
+
+    test "get_post!/2 returns the post with given id", %{user: user} do
+      %Post{id: id} = post_fixture(user)
+      assert %Post{id: ^id} = Twits.get_post!(user, id)
+    end
+
+    test "create_post/2 with valid data creates a post", %{user: user} do
+      assert {:ok, %Post{} = post} = Twits.create_post(user, @valid_attrs)
+      assert post.body == "some body"
+    end
```

Comment:
it may be helpful to check the user as well as check the database (ie call Repo.something) to check against these two cases:
1) check the user: the create_post function creates a post but makes the post author somebody other than the `user` that is passed in (unlikely but still a part of the side effect as the code includes (`put assoc`)
2) check the db: the create_post function creates simply returns a different post with same body without ever inserting into the db. In the current implementation, you are checking the return value `{:ok, post}` as evidence that the post is in the db when you are bypassing / not using the db in the test implementation.

Reply:
1. Check that the user passed in as a parameter was indeed the user assigned: `assert post.user_id == user.id`
2. check wether the post exists in the db:
`      query = from p in Post, where: p.id == ^post.id
      assert Twitter.Repo.exists?(query) == true`